src/src/Helpers/Rules_calculate.cpp

```diff
@@ -280,8 +280,13 @@ ESPEASY_RULES_FLOAT_TYPE RulesCalculate_t::apply_quinary_operator(char op,
   ESPEASY_RULES_FLOAT_TYPE ret{};
   const UnaryOperator qu_op = static_cast<UnaryOperator>(op);
 
-  if (UnaryOperator::Map == qu_op) {
-    return mapADCtoFloat(first, second, third, fourth, fifth);
+  if (UnaryOperator::Map == qu_op || UnaryOperator::MapC == qu_op) {
+    ret = mapADCtoFloat(first, second, third, fourth, fifth);
+
+    // Clamp the result if the operator is MapC
+    if (qu_op == UnaryOperator::MapC) {
+      ret = std::clamp(ret, std::min(fourth, fifth), std::max(fourth, fifth));
```

Comment:
std::clamp is not yet known in the compiler used for ESP8266.
Maybe you can use the Arduino function `constrain`